File: Legacy-Folder/Backend/HRMSWebApi/HRMS.Domain/Entities/DepartmentClearance.cs
using HRMS.Domain.Enums;

namespace HRMS.Domain.Entities
{
    public class DepartmentClearance : BaseEntity
    {

        public int ResignationId { get; set; }
        public KTStatus KTStatus { get; set; }
        public string KTNotes { get; set; }
        public string Attachment { get; set; }
        public string KTUsers { get; set; }
        public string FileOriginalName { get; set; }

    }
}

File: hrms-backend/app/Models/DepartmentClearance.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class DepartmentClearance extends Model
{
    protected $table = 'department_clearance';
    protected $primaryKey = 'Id';
    public $timestamps = false;

    protected $fillable = [
        'ResignationId', 'KTStatus', 'KTNotes', 'Attachment', 'KTUsers', 
        'FileOriginalName', 'CreatedBy', 'CreatedOn', 'ModifiedBy', 'ModifiedOn'
    ];

    protected $casts = [
        'CreatedOn' => 'datetime',
        'ModifiedOn' => 'datetime',
    ];

    // Relationship: Department Clearance belongs to Resignation
    public function resignation()
    {
        return $this->belongsTo(Resignation::class, 'ResignationId', 'Id');
    }
}
